src/index.js

```diff
@@ -1,53 +1,58 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 
-class Typographer extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { statefulGenerator: this.generator(props.text) };
-    }
-
-    componentDidMount() {
-        this.typingTimer = setInterval(
-            () => this.tick(), this.props.speed
-        );
-    }
-
-    componentWillUnmount() {
-        clearInterval(this.typingTimer);
-    }
-
-    render() {
-        var statefulGenerator = this.state.statefulGenerator;
-        return (<span> {statefulGenerator.text} </span>);
-    }
-
-    tick() {
-        this.setState({
-            statefulGenerator: this.state.statefulGenerator.next()
-        });
-    }
-
-    generator(text) {
-        return this.generatorHelper(text, 0);
-    }
-
-    generatorHelper(text, currentIndex) {
+export default class Typographer extends React.PureComponent {
+	constructor(props) {
+		super(props);
+		this.state = {
+			statefulGenerator: this.generator(props.children || props.text)
```

Comment:
We can just remove the text prop -- feel like it's an inferior feature that makes things more confusing.